analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/ProgramStateTest.HasConstraint.cs

```diff
@@ -0,0 +1,118 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarAnalyzer.SymbolicExecution;
+using SonarAnalyzer.SymbolicExecution.Constraints;
+using SonarAnalyzer.SymbolicExecution.Roslyn;
+
+namespace SonarAnalyzer.UnitTest.SymbolicExecution.Roslyn;
+
+public partial class ProgramStateTest
+{
+    private static IEnumerable<object[]> HasConstraintTestData => new[]
+    {
+        new object[] { null, ObjectConstraint.Null, false },
+        new object[] { SymbolicValue.Empty, ObjectConstraint.Null, false },
+        new object[] { SymbolicValue.Null, ObjectConstraint.Null, true },
+        new object[] { SymbolicValue.Null, ObjectConstraint.NotNull, false },
+        new object[] { SymbolicValue.Null, BoolConstraint.True, false }
+    };
+
+    [TestMethod]
+    [DynamicData(nameof(HasConstraintTestData))]
+    public void HasConstraint_Operation(SymbolicValue symbolicValue, SymbolicConstraint constraint, bool hasConstraint)
+    {
+        var cfg = TestHelper.CompileCfgBodyCS("var s = \"\";");
+        var operation = cfg.Blocks[1].Operations[0];
```

Comment:
There's already `CreateOperation()` method in the partial class that you should use.

Reply:
Same below